python/modules/IcePy/PySliceLoader.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) ZeroC, Inc.
+
+#include "PySliceLoader.h"
+#include "Types.h"
+
+#include <cassert>
+
+using namespace std;
+
+IcePy::PySliceLoader::PySliceLoader(PyObject* sliceLoader) : _sliceLoader(Py_NewRef(sliceLoader))
+{
+    assert(_sliceLoader != Py_None);
+
+    if (!PyCallable_Check(_sliceLoader))
+    {
+        throw Ice::InitializationException{__FILE__, __LINE__, "sliceLoader must be a callable"};
+    }
+}
+
+IcePy::PySliceLoader::~PySliceLoader() { Py_DECREF(_sliceLoader); }
+
+Ice::ValuePtr
+IcePy::PySliceLoader::newClassInstance(string_view typeId) const
+{
+    // The unmarshaling always runs with the GIL locked.
+    assert(PyGILState_Check());
+
+    // Get the type information.
+    ValueInfoPtr info =
```

Comment:
We can do this later, after calling the sliceLoader, this wont be used if the loader returns None.